src/main/java/org/candlepin/model/DistributorVersionCurator.java

```diff
@@ -33,66 +32,137 @@ public DistributorVersionCurator() {
+            .getResultList();
+
+        List<DistributorVersion> result = new ArrayList<>();
+        for (DistributorVersionCapability dvc : distributorVersionCapabilities) {
+            result.add(dvc.getDistributorVersion());
         }
 
-        return distVers;
+        return result;
     }
 
-    @SuppressWarnings("unchecked")
+    /**
+     * Retrieves a {@link DistributorVersion} that has an ID equal to the provided ID.
+     *
+     * @param id
+     *  the ID to retrieve a distributor version for
+     *
+     * @return a distributor version that has an ID equal to the provided ID, or null if one does not exist
+     */
     public DistributorVersion findById(String id) {
-        List<DistributorVersion> dvList = currentSession()
-            .createCriteria(DistributorVersion.class)
-            .add(Restrictions.eq("id", id)).list();
-        if (!dvList.isEmpty()) {
-            return dvList.get(0);
+        if (id == null || id.isBlank()) {
+            return null;
+        }
+
+        String query = "SELECT d FROM DistributorVersion d WHERE d.id = :id";
+
+        try {
+            return this.currentSession().createQuery(query, DistributorVersion.class)
+            .setParameter("id", id)
+            .getSingleResult();
+        }
+        catch (NoResultException e) {
+            return null;
         }
-        return null;
     }
 
-    @SuppressWarnings("unchecked")
+    /**
+     * Retrieves all of the {@link DistributorVersion}s.
+     *
+     * @return
+     *  all distributor version
+     */
     public List<DistributorVersion> findAll() {
-        return (List<DistributorVersion>) currentSession()
-            .createCriteria(DistributorVersion.class).list();
-    }
+        String query = "SELECT d FROM DistributorVersion d";
 
-    @SuppressWarnings("unchecked")
-    public Set<DistributorVersionCapability> findCapabilitiesByDistVersion(String distVersion) {
-        List<DistributorVersion> dvList = currentSession()
-            .createCriteria(DistributorVersion.class)
-            .add(Restrictions.eq("name", distVersion)).list();
-        if (!dvList.isEmpty()) {
-            return ((DistributorVersion) dvList.get(0)).getCapabilities();
-        }
-        return null;
+        return this.currentSession()
+            .createQuery(query, DistributorVersion.class)
+            .getResultList();
     }
 
+    /**
+     * Retrieves all the {@link DistributorVersionCapability} for a {@link DistributorVersion} that has a
+     * name equal to the provided name.
+     *
+     * @param distVersionName
+     *  the name of the distributor version to retrieve capabilities for
+     *
+     * @return all the capabilities for a distributor version that has a name equal to the provided name, or
+     *  null if there is no distributor version that has the provided name
+     */
+    public Set<DistributorVersionCapability> findCapabilitiesByDistVersion(String distVersionName) {
```

Comment:
This method is only used in one place and could be replaced by fetching the DV straight up and then doing the null check there. Would save maintenance and testing pain here.

Reply:
Good catch! Updated, and then updated the method that called `findCapabilitiesByDistVersion`